packages/block-editor/src/components/list-view/index.js

```diff
@@ -29,14 +29,28 @@ import { store as blockEditorStore } from '../../store';
 
 const noop = () => {};
 const expanded = ( state, action ) => {
+	if ( ! Array.isArray( action.clientIds ) ) {
+		return state;
+	}
+
+	const nextState = { ...state };
+
 	switch ( action.type ) {
-		case 'expand':
-			return { ...state, ...{ [ action.clientId ]: true } };
-		case 'collapse':
-			return { ...state, ...{ [ action.clientId ]: false } };
-		default:
-			return state;
+		case 'expand': {
+			for ( const clientId of action.clientIds ) {
+				delete nextState[ clientId ];
```

Comment:
This causes nodes without children to disappear since the visibility of `<ListViewBranch />` relies on an `isExpanded` state.

Changing it to `nextState[ clientId ] = true;` prevents this.

Also I don't see much usage of the `for ... of` statement in the clientside javascript. I guess it's because IE didn't support it. I'm not sure whether have to worry about it given this post: https://make.wordpress.org/core/2021/04/22/ie-11-support-phase-out-plan/